pkg/network/protocols/testutil/pcaputils.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package testutil
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// GetShortTestName generates a suffix in the form
+// "<protocol>-<subtest name>" to be passed to WithPCAP.
+func GetShortTestName(proto, subtest string) string {
+	subtest = strings.ReplaceAll(subtest, " ", "_")
+	return fmt.Sprintf("%s-%s", proto, subtest)
+}
+
+// WithPCAP runs tcpdump for the duration of the test.
+//
+// It returns an `io.Writer` to be used as a KeyLogWriter in
+// tls.Config to be able, to decrypt the TLS traffic in the resulting
+// PCAP. The resulting PCAPs and keylog files will be saved in
+// `/tmp/test_pcaps`
+//
+// Unless alwaysSave is true, WithPCAP will only save the resulting
+// PCAP if the test fails.
+func WithPCAP(t *testing.T, port string, suffix string, alwaysSave bool) io.Writer {
```

Comment:
It will be great to have an example (skipped) to see how to use it for TLS and for plaintext scenarios

Reply:
Added examples with Go's examples feature.